### PR TITLE
[Bug][SIMS-#1179]MSFAA and ECerts File sent bugs

### PR DESCRIPTION
**Pull request by guru-aot**
- Resolve the following issues for MSFAA and ecert send files
- [x] Change sequence number to ensure it is starting at 1 for each day (subsequent generations on the same day should be 2,3,4,etc)
- [x] Ensure the header in the file matched above change
- [x] user FELIX was not working as his firstname was NULL. Ensure students can be sent with NULL givenName. Fill with spaces
- [x] REMOVE .DAT at end of file, not actually used in send files of ESDC files(MSFAA and ECERT)

### CODE REVIEW

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts [hunk 1]
```diff
@@ -179,11 +180,11 @@ export class ECertFileHandler extends ESDCFileHandler {
           this.logger.log(`Uploading ${offeringIntensity} content...`);
           await eCertIntegrationService.uploadContent(
             fileContent,
-            fileInfo.filePath,
+            fileInfo.filePath.slice(0, -4),
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts:183]: Is this code removing the file extension? If it is there are specific libraries to do it, please check: https://nodejs.org/api/path.html#path_path_parse_path

Why do we need to remove the extension here?

File: sources/packages/api/src/esdc-integration/msfaa-integration/msfaa-files/msfaa-file-detail.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export class MSFAAFileDetail implements MSFAARequestFileLine {
     record.appendDate(this.birthDate, DATE_FORMAT);
     record.appendDate(this.processDate, DATE_FORMAT);
     record.appendWithEndFiller(this.surname, 25, SPACE_FILLER);
-    record.appendWithEndFiller(this.givenName, 15, SPACE_FILLER);
+    record.appendWithEndFiller(this.givenName || "", 15, SPACE_FILLER);
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/esdc-integration/msfaa-integration/msfaa-files/msfaa-file-detail.ts:45]: Please use the "??" that will ensure that "" will be used specifically for `null` or `undefined`.

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts (hunk 1 above)
Comment (ann-aot) [reply sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts:183]: i also has the same question

Comment (guru-aot) [reply sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts:183]: Yeah, we can do it. Thanks

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes nd fixes, looks good 👍

**Review by ann-aot — APPROVED** [review]
👍

**Review by dheepak-aot — APPROVED** [review]
LGTM.

**Review by abschwenker — APPROVED** [review]

**Pull request merged**